msdk-io/msdk-io-mzml2/src/test/java/io/github/msdk/io/mzml2/MzMLFileParserTest.java

```diff
@@ -394,7 +386,61 @@ public void testTruncated() throws MSDKException {
     File inputFile = new File(TEST_DATA_PATH + "truncated.mzML");
     Assert.assertTrue(inputFile.canRead());
     MzMLFileParser parser = new MzMLFileParser(inputFile);
+    parser.execute();
+
+  }
+
+  @Test
+  public void testZlibAndNumpressCompression() throws MSDKException {
+
+    // Import the file
+    File inputFile = new File(TEST_DATA_PATH + "MzValues_Zlib+Numpress.mzML");
+    Assert.assertTrue(inputFile.canRead());
+    MzMLFileParser parser = new MzMLFileParser(inputFile);
     RawDataFile rawFile = parser.execute();
+    Assert.assertNotNull(rawFile);
+    Assert.assertEquals(1.0, parser.getFinishedPercentage(), 0.0001);
+
+    // The file has 6 scans
+    List<MsScan> scans = rawFile.getScans();
+    Assert.assertNotNull(scans);
+    Assert.assertEquals(6, scans.size());
+
+    // 4th scan, #2103
+    MsScan scan2 = scans.get(3);
+    Assert.assertEquals(new Integer(2103), scan2.getScanNumber());
+    Assert.assertEquals(MsSpectrumType.CENTROIDED, scan2.getSpectrumType());
+    Assert.assertEquals(new Integer(1), scan2.getMsLevel());
+    Assert.assertEquals(1127.6449f, scan2.getRetentionTime(), 0.01f);
+    Assert.assertEquals(PolarityType.NEGATIVE, scan2.getPolarity());
+    scan2.getMzValues();
```

Comment:
In fact, the check of a few values should be added to all the tests that call getMzValues();
These checks were missing in the old tests
Also, please add a similar checks to getIntensityValues()